moveit_ros/visualization/motion_planning_rviz_plugin/include/moveit/motion_planning_rviz_plugin/motion_planning_display.h

```diff
@@ -295,6 +298,9 @@ private Q_SLOTS:
   rviz::BoolProperty* query_start_state_property_;
   rviz::BoolProperty* query_goal_state_property_;
   rviz::FloatProperty* query_marker_scale_property_;
+  rviz::Property* query_marker_offset_category_;
+  rviz::VectorProperty*  query_marker_offset_position_property_;
+  rviz::QuaternionProperty* query_marker_offset_orientation_property_;
```

Comment:
These are allocated on the heap, where is the lifetime of them being managed (where are they deallocated)?

Reply:
This is handled by Qt automatically via parent-child relations. A parent QObject will delete all of it's children upon destruction. Children will sign off at their parent if they are destroyed before.